lib/workers/global/config/parse/file.ts

```diff
@@ -18,7 +19,12 @@ export async function getParsedContent(file: string): Promise<RenovateConfig> {
       return JSON5.parse(await readFile(file, 'utf8'));
     case '.js': {
       const tmpConfig = await import(file);
-      return tmpConfig.default ? tmpConfig.default : tmpConfig;
+      let config = tmpConfig.default ? tmpConfig.default : tmpConfig;
+      // Allow the config to be a function
+      if (is.fn(config)) {
+        config = config();
+      }
+      return config;
```

Comment:
@viceice Are you sure? The signature of the function containing the code is:

    export async function getParsedContent(file: string): Promise<RenovateConfig> {}

If typescript is like javascript, won't getParsedContent() return a rejected `Promise` if an exception is thrown inside the function, regardless of whether it is awaited or not? See this snippet:

```bash
$ node -e '(async function() { throw new Error("yup") })().catch(error => { console.log("got error", error) })'
got error Error: yup
    at [eval]:1:27
    at [eval]:1:46
    at Script.runInThisContext (node:vm:129:12)
    at Object.runInThisContext (node:vm:305:38)
    at node:internal/process/execution:75:19
    at [eval]-wrapper:6:22
    at evalScript (node:internal/process/execution:74:60)
    at node:internal/main/eval_string:27:3
```

If you want me to, I'll change the line to:

            config = await config();

but unless I'm enlightened with new knowledge, I can't think of a situation where it will make any difference.

Reply:
You are right, the catch is in calling function, so leave as is.

Reply:
But there _is_ a related problem: Regardless of what exception `config.js` throws or what problems it encounters, `file.ts` causes this:

```bash
# Deliberately introduce an error in config.js
$ git diff lib/workers/global/config/parse/__fixtures__/config.js
diff --git a/lib/workers/global/config/parse/__fixtures__/config.js b/lib/workers/global/config/parse/__fixtures__/config.js
index e4fe0d854..7a1c443a5 100644
--- a/lib/workers/global/config/parse/__fixtures__/config.js
+++ b/lib/workers/global/config/parse/__fixtures__/config.js
@@ -1,4 +1,5 @@
 // @ts-ignore
+throw new Error("foo");
 module.exports = {
   token: 'abcdefg',
 };

$ RENOVATE_CONFIG_FILE=lib/workers/global/config/parse/__fixtures__/config.js yarn start
yarn run v1.22.15
$ run-s generate:* 
$ node tools/generate-imports.mjs
generating imports
> data/node-js-schedule.json
> node_modules/emojibase-data/en/shortcodes/github.json
$ node -r ts-node/register/transpile-only -- lib/renovate.ts
FATAL: No custom config file found on disk
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The error `FATAL: No custom config file found on disk` is clearly wrong. And worse, the exception containing the error that be really helpful to the poor user is nowhere to be found.

Reply:
This can be fixed in a seperate PR